Userland/Libraries/LibWeb/Layout/FlexFormattingContext.cpp

```diff
@@ -612,13 +614,21 @@ void FlexFormattingContext::determine_flex_base_size_and_hypothetical_main_size(
                 return false;
             if (flex_basis.length_percentage->is_length())
                 return true;
+
+            bool can_resolve_percentages = is_row_layout()
+                ? m_flex_container_state.has_definite_width()
+                : m_flex_container_state.has_definite_height();
+
             if (flex_basis.length_percentage->is_calculated()) {
-                // FIXME: Handle calc() in used flex basis.
+                auto const& calc_value = *flex_basis.length_percentage->calculated();
+                if (calc_value.resolves_to_length())
+                    return true;
+                if (calc_value.resolves_to_percentage())
+                    return can_resolve_percentages;
```

Comment:
The current calc code is a bit ad-hoc, and so this won't work quite the way you think.

`calc(5px + 50%)` gets a resolved type of "length", which is wrong but we don't implement the proper algorithm yet. You can check if there's a percentage in there with `contains_percentage()` though.

So, something more like:

```suggestion
                if (calc_value.resolves_to_percentage() 
                    || (calc_value.resolves_to_length() && calc_value.contains_percentage()))
                    return can_resolve_percentages;
                if (calc_value.resolves_to_length())
                    return true;
```

Reply:
Oh all right, let's do that then :) Thanks!